rmw_fastrtps_shared_cpp/src/rmw_request.cpp

```diff
@@ -102,8 +102,9 @@ __rmw_take_request(
     // Get header
     rmw_fastrtps_shared_cpp::copy_from_fastrtps_guid_to_byte_array(
       request.sample_identity_.writer_guid(),
-      request_header->writer_guid);
-    request_header->sequence_number = ((int64_t)request.sample_identity_.sequence_number().high) <<
+      request_header->request_id.writer_guid);
+    request_header->request_id.sequence_number =
+      ((int64_t)request.sample_identity_.sequence_number().high) <<
```

Comment:
Just wondering, shouldn't `source_timestamp` be set here? I didn't follow the design/implementation of this, so I really don't know.

There's a test that is sometimes failing in nightlies https://ci.ros2.org/view/nightly/job/nightly_osx_repeated/1939/testReport/junit/rclpy.src.ros2.rclpy.rclpy.test.test_client/TestClient/test_service_timestamps/, that's why I'm asking.
Line where the test is failing: https://github.com/ros2/rclpy/blob/f1a09c20a4af221da7db476206fea48a83284db7/rclpy/test/test_client.py#L109

Reply:
It should...

Reply:
@iluetkeb can you provide a fix?

Another interesting comment, the test I linked wasn't always failing.
AFAIS, it only failed sometimes in macOS.

It may be the case that this `struct` doesn't have a default initialization, and that's why it didn't always fail.
In that case, we should add a default initialization too.

Reply:
Right on both counts. I added a default init for the message_info changes, but not here. I'll address that together.

Reply:
https://github.com/ros2/rmw_fastrtps/pull/378 and https://github.com/ros2/rcl/pull/642 address this.